app/models/reservation.rb

```diff
@@ -165,22 +169,26 @@ def admin?
     order.nil? && !blackout?
   end
 
+  def admin_removable?
+    true
+  end
+
   def blackout?
     blackout.present?
   end
 
   def can_start_early?
     return false unless in_grace_period?
     # no other reservation ongoing; no res between now and reserve_start;
-    where = <<-SQL
-      reserve_start_at > ?
-      AND reserve_start_at < ?
-      AND actual_start_at IS NULL
-      AND reservations.product_id = ?
-      AND (order_detail_id IS NULL OR order_details.state = 'new' OR order_details.state = 'inprocess')
-    SQL
-
-    Reservation.joins(:order_detail).where(where, Time.zone.now, reserve_start_at, product_id).first.nil?
+
+    Reservation
+      .not_started
+      .where("reserve_start_at > :now", now: Time.current)
+      .where("reserve_start_at < :reserve_start_at", reserve_start_at: reserve_start_at)
+      .where(product_id: product_id)
+      .joins(:order_detail)
+      .where("order_detail_id IS NULL OR order_details.state IN ('new', 'inprocess')")
```

Comment:
The existing `new_or_inprocess` scope unfortunately comes with baggage. It looks like it's only used in 3 spots (outside specs) so it might not be too painful to clean up.